tools/analyzer_plugin/test/unit/util/boilerplate_utils_test.dart

```diff
@@ -0,0 +1,282 @@
+          removeOverReactGeneratedPartDirective(builder, result.unit);
+        });
+        final editList = sourceChange.edits?.firstOrNull?.edits;
+
+        expect(editList, isNotNull);
+        expect(editList.length, 1, reason: 'there should be one edit in the file');
+
+        final part = getOverReactGeneratedPartDirective(result.unit);
+        expect(editList.first.offset, part.offset);
+        expect(editList.first.length, part.length);
+        expect(editList.first.replacement, '');
+      });
+    });
+
+    group('fixOverReactGeneratedPartDirective', () {
+      group('does nothing if', () {
+        test('there is no part directive in file', () async {
+          final sourceFileEdits = await getSourceFileEdits(
+            sourceWithNoPart,
+            (builder, result) => fixOverReactGeneratedPartDirective(builder, result.unit, result.uri),
+            path: 'foo.dart',
+          );
+          expect(sourceFileEdits, isEmpty);
+        });
+
+        test('part directive in file is valid', () async {
+          final sourceFileEdits = await getSourceFileEdits(
+            sourceWithOverReactPart,
+            (builder, result) => fixOverReactGeneratedPartDirective(builder, result.unit, result.uri),
+            path: 'foo.dart',
+          );
+          expect(sourceFileEdits, isEmpty);
+        });
+
+        test('part directive in file is not over_react', () async {
+          final sourceFileEdits = await getSourceFileEdits(
+            sourceWithNonOverReactPart,
+            (builder, result) => fixOverReactGeneratedPartDirective(builder, result.unit, result.uri),
+            path: 'foo.dart',
+          );
+          expect(sourceFileEdits, isEmpty);
+        });
+      });
+
+      test('replaces over_react part directive that does not match file uri', () async {
+        final result = await parseAndGetResolvedUnit(sourceWithOverReactPart, path: 'different_file_name.dart');
+        final sourceChange = await buildFileEdit(result, (builder) {
+          fixOverReactGeneratedPartDirective(builder, result.unit, result.uri);
+        });
+        final editList = sourceChange.edits?.firstOrNull?.edits;
+
+        expect(editList, isNotNull);
+        expect(editList.length, 1, reason: 'there should be one edit in the file');
+
+        final part = getOverReactGeneratedPartDirective(result.unit);
+        expect(editList.first.offset, part.offset);
+        expect(editList.first.length, part.length);
+        expect(editList.first.replacement, 'part \'different_file_name.over_react.g.dart\';');
+      });
+    });
+  });
+}
+
+Future<void> overReactGeneratedPartDirectiveIsValidTestHelper({
+  bool isPartOverReact,
+  String path,
+  bool expectedResult,
+}) async {
+  final result = await parseAndGetResolvedUnit(
+    isPartOverReact ? sourceWithOverReactPart : sourceWithNonOverReactPart,
+    path: path,
+  );
+  final part = result.unit.directives.whereType<PartDirective>()?.first;
+  expect(part, isNotNull);
```

Comment:
I could definitely be wrong, but I don't believe this test will ever fail directly. I don't believe `whereType` would ever return `null`, instead it returns an empty list. Then `.first` would throw a state error. This might fine since it still fails the test, but `firstOrNull` or `firstWhere` with a type check would have the behavior I'd expect 267 to be checking for

Reply:
Good catch! I'll update it to `firstOrNull`